cpp/src/arrow/util/secure_string.cc

```diff
@@ -0,0 +1,147 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#if defined(ARROW_USE_OPENSSL)
+#  include <openssl/crypto.h>
+#  include <openssl/opensslv.h>
+#endif
+#include <utility>
+#if defined(_WIN32)
+#  include <windows.h>
+#endif
+
+#include "arrow/util/secure_string.h"
+#include "arrow/util/span.h"
+
+namespace arrow::util {
+SecureString::SecureString(SecureString&& secret) noexcept
+    : secret_(std::move(secret.secret_)) {
+  secret.Dispose();
+}
+SecureString::SecureString(std::string&& secret) noexcept : secret_(std::move(secret)) {
```

Comment:
As a nit, can you leave newlines between the various definitions? For example:
```suggestion
namespace arrow::util {

SecureString::SecureString(SecureString&& secret) noexcept
    : secret_(std::move(secret.secret_)) {
  secret.Dispose();
}

SecureString::SecureString(std::string&& secret) noexcept : secret_(std::move(secret)) {
```

Reply:
sure